### PR TITLE
Add code field to request body (PUT /me)

### LINKED ISSUE #205: Require code during special account update
### Front 
Impacted component:

- `special-account-edit`

### Back:

Impacted endpoints:
- `PUT /me`
- `PUT /special-accounts/{id}`

### PR DESCRIPTION
**Pull request by kistora**
### Description of your PR

Will close #205 
I also change a bit the layout

### Pre-review TODO

- [ ] PR is rebased on top of `origin/master` 
- [ ] PR provide new tests for new behaviors
- [ ] PR doesn't introduce commented out code
- [ ] PR has been tested locally on the main use cases
- [ ] Documentation have been updated if necessary

### CODE REVIEW

File: client/src/app/_services/me.service.ts [hunk 1]
```diff
@@ -8,14 +8,15 @@ export class MeService {
 
   constructor(private http: HttpClient) { }
 
-  put(connectedUser: ConnectedUser): Observable<any> {
+  put(connectedUser: ConnectedUser, code?: Number): Observable<any> {
```
Comment (Embraser01) [inline client/src/app/_services/me.service.ts]: Why optionnal?

Comment (kistora) [reply client/src/app/_services/me.service.ts]: For barman update :)

**Comment by kistora** [comment]
Oh you are right, I misunderstood the issue :) I will re work on it tomorrow

**Comment by kistora** [comment]
The middleware is already implemented:
- `routes/special-account.js` throught `am(codeGuard)` for `PUT /specialaccounts/{id}`
- `controllers/me.js` for `PUT /me`

**Pull request merged**